File: test_quran.py
import re

def removeTashkeel(text):
    return re.sub(r'[\u0617-\u061A\u064B-\u0652\u0670]', '', text)

def normalizeArabic(text):
    text = removeTashkeel(text)
    text = re.sub(r'[أإآٱ]', 'ا', text)
    text = text.replace("ة", "ه")
    text = text.replace("ى", "ي")
    text = text.replace("ؤ", "و")
    text = text.replace("ئ", "ي")
    return text

text = "بِسْمِ ٱللَّهِ ٱلرَّحْمَٰنِ ٱلرَّحِيمِ"
query = "ٱلرَّحْمَٰنِ"
print("Text:", normalizeArabic(text))
print("Query:", normalizeArabic(query))

File: test_quran.kt
import java.io.File

fun main() {
    val text = "بِسْمِ ٱللَّهِ ٱلرَّحْمَٰنِ ٱلرَّحِيمِ"
    val query = "ٱلرَّحْمَٰنِ"
    
    fun removeTashkeel(t: String): String {
        return t.replace(Regex("[\\u0617-\\u061A\\u064B-\\u0652\\u0670]"), "")
    }
    fun normalizeArabic(t: String): String {
        return removeTashkeel(t)
            .replace(Regex("[أإآٱ]"), "ا")
            .replace("ة", "ه")
            .replace("ى", "ي")
            .replace("ؤ", "و")
            .replace("ئ", "ي")
    }
    
    println(normalizeArabic(text))
    println(normalizeArabic(query))
}
